### ISSUE TITLE
D7 dev/stage sites not disallowed in robots.txt

### ISSUE THREAD

**Issue by pyrello** [uiowa/uiowa#5466]
What should our long term solution be for this?

**Comment by briand44** [comment]
We could possibly use https://api.druphelp.com/api/robotstxt/robotstxt.api.php/function/hook_robotstxt/7 in our profiles to check for environment variables and add disallow: /* in dev and stage environments. I am not sure adding that to what already exists is proper robots.txt markup or not

**Comment by mjoneill** [comment]
https://support.google.com/webmasters/answer/6062598?hl=en

Google's robots.txt tester

**Comment by joewhitsitt** [comment]
Low-tech suggestion: Could we just clean up dev/stage sites on d7 again and manually change the robots.txt to disallow each time we need to sync a site for a period of time? Recently we have only synced to dev for migration testing or to debug an issue

**Comment by briand44** [comment]
>Low-tech suggestion: Could we just clean up dev/stage sites on d7 again and manually change the robots.txt to disallow each time we need to sync a site for a period of time? Recently we have only synced to dev for migration testing or to debug an issue

Not opposed to that but not sure what it will take to clean up dev/stage sites? There are a lot of standard sites.

**Comment by joewhitsitt** [comment]
Yes there are a lot, but they aren't being used and this would have the added benefit of freeing up server resources
